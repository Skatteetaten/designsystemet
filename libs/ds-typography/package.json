{
  "name": "@skatteetaten/ds-typography",
  "version": "1.1.0",
  "groupId": "no.skatteetaten.aurora",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.1.0"
  },
  "scripts": {}
}
