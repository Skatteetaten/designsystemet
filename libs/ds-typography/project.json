{
  "name": "ds-typography",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-typography/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-typography",
        "tsConfig": "libs/ds-typography/tsconfig.lib.json",
        "project": "libs/ds-typography/package.json",
        "entryFile": "libs/ds-typography/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-typography/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-typography/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-typography/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-typography/**/*.css",
          "libs/ds-typography/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-typography"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish  dist/libs/ds-typography {args.switches}"
      }
    }
  }
}
