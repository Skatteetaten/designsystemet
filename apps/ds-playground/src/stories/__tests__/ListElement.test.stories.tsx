import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { List } from '@skatteetaten/ds-typography';

const meta = {
  component: List.Element,
  title: 'Tester/List/Element',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: 'text',
    },
  },
} satisfies Meta<typeof List.Element>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs = {
  children: 'Kjenner du behovet til brukeren?',
};
const Template: StoryFn<typeof List.Element> = (args) => (
  <ul>
    <List.Element {...args} />
  </ul>
);

export const WithRef = {
  render: Template,
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLLIElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const list = canvas.getByRole('listitem');
    await expect(list).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  render: Template,
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const list = canvas.getByRole('listitem');
    await expect(list).toHaveAttribute('id', 'htmlid');
    await expect(list).toHaveClass('dummyClassname');
    await expect(list).toHaveAttribute('lang', 'nb');
    await expect(list).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;
