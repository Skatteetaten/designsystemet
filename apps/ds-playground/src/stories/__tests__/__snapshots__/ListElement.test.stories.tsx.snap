// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/List/Element WithAttributes play-test 1`] = `
<div data-test-block="true">
  <ul>
    <li id="htmlid"
        class="_element_bxsik_1 dummyClassname"
        lang="nb"
        data-testid="123ID"
    >
      Kjenner du behovet til brukeren?
    </li>
  </ul>
</div>
`;

exports[`Tester/List/Element WithRef play-test 1`] = `
<div data-test-block="true">
  <ul>
    <li class="_element_bxsik_1 "
        id="dummyIdForwardedFromRef"
    >
      Kjenner du behovet til brukeren?
    </li>
  </ul>
</div>
`;
